File: PAT_Basic_Level_by_cpp/1067 试密码/1067 试密码.cpp
// 1067 试密码 - 达到 N 次错误输出 Account locked 并结束，处理 # 结束符
#include <iostream>
#include <string>
using namespace std;
int main() {
    string pwd;
    int n;
    if (!(cin >> pwd >> n)) return 0;
    string dummy;
    getline(cin, dummy); // 消耗行尾换行
    int cnt = 0;
    string line;
    while (getline(cin, line)) {
        if (line == "#") break;
        cnt++;
        if (line == pwd) {
            cout << "Welcome in" << endl;
            break;
        } else {
            cout << "Wrong password: " << line << endl;
            if (cnt >= n) {
                cout << "Account locked" << endl;
                break;
            }
        }
    }
    return 0;
}

File: PAT_Basic_Level_by_r/1067 试密码/1067 试密码.R
# 题目名称：试密码
# 题目编号：PAT Basic 1067
# 实现原理：
#   逐行读取用户输入的密码尝试，与正确密码比较。
#   如果输入为"#"则结束输入。密码正确输出欢迎信息，
#   错误则输出错误提示。
# 时间复杂度：O(N*L)，N为尝试次数，L为密码长度
# 空间复杂度：O(1)，仅存储正确密码
x<-readLines("stdin",warn=FALSE);p<-strsplit(x[1],"\\s+")[[1]];pass<-p[1];n<-as.integer(p[2]);bad<-0L;for(s in x[-1]){if(s=="#")break;if(s==pass){cat("Welcome in\n");break}else{bad<-bad+1L;cat("Wrong password: ",s,"\n",sep="");if(bad==n){cat("Account locked\n");break}}}
